src/main/java/com/xceptance/xlt/engine/htmlunit/okhttp3/AuthenticatorImpl.java

```diff
@@ -35,11 +38,51 @@
 
 /**
  * An OkHttp {@link Authenticator} that is backed by HtmlUnit's credentials provider.
+ *
+ * @see AuthorizationHeaderInterceptor
  */
 class AuthenticatorImpl implements Authenticator
 {
+    /**
+     * The result of an authentication attempt ready to be stored in the authentication cache.
+     */
+    static class AuthResult
```

Comment:
*LOW* Looks like this is a perfect candidate for being implemented as record. WDYT?